File: mrshudson-backend/src/main/java/com/mrshudson/domain/dto/SendMessageRequest.java
package com.mrshudson.domain.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

/**
 * 发送消息请求DTO
 */
@Data
public class SendMessageRequest {

    /**
     * 用户消息内容
     */
    @NotBlank(message = "消息内容不能为空")
    private String message;

    /**
     * 会话ID（可选，用于多轮对话）
     */
    private String sessionId;

    /**
     * 对话会话ID（可选，用于关联会话）
     */
    private Long conversationId;
}

File: mrshudson-android/app/src/main/java/com/mrshudson/android/data/remote/dto/SendMessageRequest.kt
package com.mrshudson.android.data.remote.dto

/**
 * 发送消息请求数据类
 *
 * @property message 消息内容
 * @property sessionId 会话ID（可选）
 * @property conversationId 会话ID，可选，为空时创建新会话
 */
data class SendMessageRequest(
    val message: String,
    val sessionId: String? = null,
    val conversationId: Long? = null
)
